sources/packages/backend/apps/api/src/institution-integration/ier-integration/ier12-file.service.ts

```diff
@@ -0,0 +1,172 @@
+import { Injectable } from "@nestjs/common";
+import { StudentAssessment } from "@sims/sims-db";
+import { InjectLogger } from "../../common";
+import { LoggerService } from "../../logger/logger.service";
+import { ConfigService, StudentAssessmentService } from "../../services";
+import { InstitutionIntegrationConfig } from "../../types";
+import { getFileNameAsCurrentTimestamp } from "../../utilities";
+import { IER12IntegrationService } from "./ier12-integration.service";
+import {
+  IER12Record,
+  IER12UploadResult,
+} from "./models/ier12-integration.model";
+
+@Injectable()
+export class IER12FileService {
+  InstitutionIntegrationConfig: InstitutionIntegrationConfig;
+  constructor(
+    config: ConfigService,
+    private readonly ierIntegrationService: IER12IntegrationService,
+    private readonly studentAssessmentService: StudentAssessmentService,
+  ) {
+    this.InstitutionIntegrationConfig =
+      config.getConfig().InstitutionIntegrationConfig;
+  }
+
+  /**
+   * 1. Fetches the assessment data for the institution location.
+   * 2. Create the Request content for the IER 12 file by populating the
+   * header, footer and trailer content.
+   * 3. Create the request filename with the file path with respect to the institution code
+   * for the IER 12 Request sent File.
+   * 4. Upload the content to the zoneB SFTP server.
+   * @param generatedDate date in which the assessment for
+   * particular institution is generated.
+   * @returns Processing IER 12 request result.
+   */
+  async processIER12File(generatedDate?: Date): Promise<IER12UploadResult[]> {
+    this.logger.log(
+      `Retrieving pending assessment on ${generatedDate} for IER 12 request...`,
```

Comment:
for optional date, `generatedDate` is null right? is that what we are looking for in this log